addons/sale_stock/models/sale_order.py

```diff
@@ -112,6 +112,11 @@ def write(self, values):
                         You should probably update the partner on this document.""") % addresses
                 picking.activity_schedule('mail.mail_activity_data_warning', note=message, user_id=self.env.user.id)
 
+        if values.get('commitment_date'):
+            # protagate commitment_date as the deadline of the related stock move.
+            # TODO: Log a note on each down document
+            self.order_line.move_ids.date_deadline = values.get('commitment_date')
```

Comment:
cast string in date (see https://github.com/odoo/odoo/commit/8f122297f80e06c06a3a5efc6815b9ba627d0ed8)